examples/C++/DDS/CustomListenerExample/CustomListenerSubscriber.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomListenerSubscriber.cpp
+ *
+ */
+
+#include "CustomListenerSubscriber.h"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/SubscriberAttributes.h>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+
+#include <thread>
+
+using namespace eprosima::fastdds::dds;
+
+CustomListenerSubscriber::CustomListenerSubscriber()
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+}
+
+bool CustomListenerSubscriber::init(bool use_dr)
+{
+    DomainParticipantQos pqos;
+    pqos.name("Participant_sub");
+    p_mask_ = StatusMask::all();
+    p_mask_.set(9,false);
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(0, pqos, &p_listener_, p_mask_);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    //CREATE THE SUBSCRIBER
+    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+
+    if (subscriber_ == nullptr)
+    {
+        return false;
+    }
+
+    //CREATE THE TOPIC
+    topic_ = participant_->create_topic(
+            "HelloWorldTopic",
+            "HelloWorld",
+            TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE READER
+    StatusMask none_mask = StatusMask::none();
+    StatusMask all_mask = StatusMask::all();
+    dr_mask_ = use_dr ? all_mask : none_mask;
+    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+    rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    reader_ = subscriber_->create_datareader(topic_, rqos, &dr_listener_, dr_mask_);
```

Comment:
Same as with the publisher
```suggestion
    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
    rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
    reader_ = subscriber_->create_datareader(topic_, rqos, &dr_listener_,
             use_dr ? StatusMask::all() : StatusMask::none());
```